wagtail/wagtailadmin/forms.py

```diff
@@ -287,6 +287,24 @@ def clean_seo_title(self):
             return self.cleaned_data['seo_title'].strip()
 
     def clean(self):
+
+        # Check for dupe form field labels - fixes #585
+        if 'form_fields' in self.formsets:
```

Comment:
Instead of changing the form used for all Pages, you can make a custom Wagtail admin form for Form Pages, and tell Wagtail to use that form when validating those page types:

```python
# in wagtail.wagtailforms.forms
class WagtailAdminFormPageForm(WagtailAdminPageForm):
    def clean(self):
        super(WagtailAdminFormPageForm, self).clean()
        # Your tests
```

```python
# in wagtail.wagtailforms.models
from .forms import WagtailAdminFormPageForm
class AbstractForm(Page):
    # ...
    # Use the custom form defined above to validate Form pages
    base_form_class = WagtailAdminFormPageForm
```